app/core/interfaces/numeric/index.js

```diff
@@ -46,18 +32,21 @@ define([
           value = this.options.schema.get('default_value');
       }
 
+      var step = 1;
+
+      if (this.options.schema.get('type') === 'FLOAT' || this.options.schema.get('type') === 'VARCHAR') {
```

Comment:
CHAR (and VARCHAR) will be saved as string and can therefor accept any step. I've added CHAR to the check as well in commit d176fe3d26c78e8bbbbcd8cd764e36f8816c9849